event_project/views/event_event_view.xml

```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0"?>
+<openerp>
+  <data>
+    <record id="view_event_event_form" model="ir.ui.view">
+      <field name="name">view_event_event_form</field>
+      <field name="model">event.event</field>
+      <field name="inherit_id" ref="event.view_event_form"/>
+      <field name="arch" type="xml">
+
+        <field name="date_tz" position="after">
+            <field name="project_related" attrs="{'invisible':[('project_template','=',False)]}"/>
+            <field name="project_template" attrs="{'readonly':[('project_related','!=',False)],'invisible':[('project_related','!=',False)]}"/>
```

Comment:
Don't make it invisible. You're already preventing its change with readonly flag.